### PR TITLE
[ETL-434] Modify S3 to Glue Lambda code to use S3 event notifications

### PR DESCRIPTION
**Pull request by philerooski**
Effectively a rewrite of our S3 to Glue Lambda to use S3 event notifications. We skirt the owner.txt issues identified in the JIRA ticket by ignoring any objects which contain "owner.txt" in their name.

### CODE REVIEW

File: src/lambda_function/s3_to_glue/app.py [hunk 1]
```diff
@@ -86,20 +50,19 @@ def lambda_handler(event, context):
...
-    trigger_event_date = parser.isoparse(event["time"]).date()
-    objects = s3.list_objects_v2(Bucket=source_bucket_name)
-    files_to_submit = []
-    if "Contents" in objects:
-        files_to_submit = query_files_to_submit(
-            objects, files_to_submit, trigger_event_date
+    messages = []
+    for record in event["Records"]:
+        bucket_name = record["s3"]["bucket"]["name"]
+        object_key = record["s3"]["object"]["key"]
```
Comment (thomasyu888) [inline src/lambda_function/s3_to_glue/app.py:56]: Just a note here: https://docs.aws.amazon.com/AmazonS3/latest/userguide/notification-content-structure.html.  It was helpful to know the S3 event notification structure.

One thing I'm not fully understanding is that this seems to just take in all the new records within a bucket without specific folders.  Based on what you were saying, am I understanding correctly that we would set up S3 notifications specific to different folders?

**Review by thomasyu888 — COMMENTED** [review]
Nice! just a question prior to approval

File: src/lambda_function/s3_to_glue/app.py (hunk 1 above)
Comment (philerooski) [reply src/lambda_function/s3_to_glue/app.py:56]: The trigger can be configured to only alert for objects under a specific prefix, e.g., `{namespace}/raw_data/...`

**Review by rxu17 — APPROVED** [review]
Lgtm!

File: tests/test_s3_to_glue_lambda.py [hunk 2]
```diff
@@ -1,108 +1,70 @@
...
+        return messages
 
-def test_empty_bucket_contents(test_trigger_event):
-    test_submit_files = app.query_files_to_submit(
-        objects={"Contents": []},
-        files_to_submit=[],
-        trigger_event_date=parser.isoparse(test_trigger_event["time"]).date(),
-    )
-    assert test_submit_files == []
+    def test_submit_s3_to_json_workflow(self, test_messages, monkeypatch):
```
[the remark this answers is not in the document]
Comment (philerooski) [reply tests/test_s3_to_glue_lambda.py:65]: I wanted to use moto but moto does not support most methods of the Glue client.

**Pull request merged**